### DISCUSSION TITLE
Surface Pro 3 - What should I notice?

### DISCUSSION THREAD

**Discussion by Discostu36** [linux-surface/linux-surface#612] [General]
Well, this might be kind of a stupid question but I just updated MX Linux KDE from Kernel 5.10 to 5.14.13 Surface and didn't notice anything different. Did I miss anything?

**Comment by qzed** [comment]
Not much. Almost(?) everything for the Surface Pro 3 should already be upstream for quite some time.

I'm not entirely sure on the WiFi situation. I know that later generations (specifically Pro 4, 5 and 6) have a ton of WiFi issues regarding their Marvell chip for which fixes are making their way upstream (thanks to @jonas2515) just now (I think some may already have been backported or might soon-ish be). IIRC the Pro 3 also has a Marvell WiFi chip, but I think that might be a different generation (so may have no or less issues if you're lucky).

In short, if you don't have any WiFi issues on the v5.10 kernel, you shouldn't notice any difference. If you do have some, WiFi with the Surface kernel should (hopefully) work better.
